### PR TITLE
Internal: Improved version control mechanism [ED-10790]

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: includes/settings/tools.php [hunk 1]
```diff
@@ -212,6 +215,11 @@ private function get_rollback_versions() {
 				return [];
 			}
 
+			$plugin_information->versions = apply_filters(
```
Comment (Nevoss) [inline includes/settings/tools.php:218]: If there something returns from the filter there is no need to call WordPress API up there `$plugin_information = plugins_api(`

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: includes/settings/tools.php (hunk 1 above)
Comment (ManorHazazE) [reply includes/settings/tools.php]: done

File: includes/settings/tools.php [hunk 2]
```diff
@@ -197,32 +208,41 @@ public function __construct() {
...
 			$current_index = 0;
-			foreach ( $plugin_information->versions as $version => $download_link ) {
+			foreach ( $versions as $version ) {
 				if ( $max_versions <= $current_index ) {
 					break;
 				}
 
+				if ( substr( $version, 0, 1 ) == 'v' ) {
+					$version = substr( $version, 1 );
+				}
```
Comment (Nevoss) [inline includes/settings/tools.php:244]: Why needed?

File: includes/settings/tools.php [hunk 3]
```diff
@@ -197,32 +208,41 @@ public function __construct() {
...
+					]
+				);
+
+				if ( empty( $plugin_information->versions ) || ! is_array( $plugin_information->versions ) ) {
+					return [];
+				}
 
-			uksort( $plugin_information->versions, 'version_compare' );
-			$plugin_information->versions = array_reverse( $plugin_information->versions );
+				uksort( $plugin_information->versions, 'version_compare' );
```
Comment (Nevoss) [inline includes/settings/tools.php:230]: Are you bringing the versions in the right order?

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: includes/settings/tools.php (hunk 2 above)
Comment (ManorHazazE) [reply includes/settings/tools.php:244]: removed

File: includes/settings/tools.php (hunk 3 above)
Comment (ManorHazazE) [reply includes/settings/tools.php:230]: YES

**Review by Nevoss — APPROVED** [review]

**Pull request merged**